src/main/java/org/galatea/starter/service/AlphaVantageService.java

```diff
@@ -0,0 +1,92 @@
+package org.galatea.starter.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.Iterator;
+import java.util.TreeMap;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.sf.aspect4log.Log;
+import org.galatea.starter.MyProps;
+import org.galatea.starter.domain.MongoDocument;
+import org.galatea.starter.domain.rpsy.StockPriceRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@RequiredArgsConstructor
+@Slf4j
+@Log
+@Service
+public class AlphaVantageService {
+
+  @Autowired
+  RestTemplate restTemplate;
+
+  @Autowired
+  StockPriceRepository stockPriceRepository;
+
+  @NonNull
+  private ObjectMapper objectMapper;
+
+  /**
+   * Serves StockPriceController, handling either the calls to database or request to AlphaVantage.
+   * @param symbol String, stock symbol user wants to find the prices of
+   * @param days int, the number of days of stock price data to return
+   * @return a String, gross mashup of proper JSON {in process of fixing}
+   */
+  public TreeMap<String,MongoDocument> access(final String symbol, final int days) {
+    String alphaVantageUrl
+        = "https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol=" + symbol;
+    String output = days > 100 ? "full" : "compact";
+    String requestUrl = alphaVantageUrl + "&outputsize=" + output + "&apikey=" + MyProps.apiKey;
+    ResponseEntity<String> response = restTemplate.getForEntity(requestUrl, String.class);
```

Comment:
Consider mapping this to an intermediate object, or directly to MongoDocument instead of getting a String.class and then doing the conversion manually with mapJsonGraph()